### DISCUSSION TITLE
Lolin S2 Mini not recognised as MIDI-Device

### DISCUSSION THREAD

**Discussion by yani66** [tttapa/Control-Surface#1033] [Q&A] [answered]
Hi,

I'm building a MIDI-Controller to use with VoiceMeeter.

I had it working with a Teensy 4.0 but had change the design to an Lolin S2 Mini due to reasons. I can upload with no problems (testet with blinking LED), but neither with a MIDI over USB nor a serial connection I can get any information of my faders.

In the Serial Monitor of the ArduinoIde are only cryptic symbols when programed to Serial (testet multiple baud rates).

Every PC I plug it in to it get's recognised as Lolin S2 Mini, but it should be called MIDI-Ghost, because I set it like that in the library.

My code (for MIDI over USB) is just the [example](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d8d/Multiple-Control-Change-Potentiometers_8ino-example.html).
For the serial connection I used another [example](https://tttapa.github.io/Control-Surface-doc/Doxygen/dd/ddd/Serial-Interface_8ino-example.html), there are three possibilities for a serial connection and I tried every one.

Thanks for any help or input.

yani66

**Comment by tttapa** [comment]
I've just enabled ESP32-S2 support in b12db8cc634c0ae0a2ca5460d1d027220109f2ab. 

> In the Serial Monitor of the ArduinoIde are only cryptic symbols when programed to Serial (testet multiple baud rates).

This is the expected behavior: MIDI data is not text, it's not readable using the serial monitor. If you want to print the MIDI messages in a readable format, use one of the "debug" MIDI interfaces from https://tttapa.github.io/Control-Surface/Doxygen/dc/df0/group__MIDIInterfaces.html.

> Every PC I plug it in to it get's recognised as Lolin S2 Mini, but it should be called MIDI-Ghost, because I set it like that in the library.

Where did you set this?

**Reply by yani66** [reply]
I updated the library, but none of my programs recognise it as a MIDI-Device.


> Where did you set this?

I'm sorry, I confused this with the [Teensy_USB-MIDI-name.py](https://github.com/tttapa/Control-Surface/blob/main/src/MIDI_Interfaces/Teensy-USB-MIDI-name.py) file.

**Reply by tttapa** [reply, marked as answer]
Are you using the latest pre-release version of the esp32-arduino core?

If you're using Windows, https://github.com/tttapa/Control-Surface/discussions/1004 might be useful.

**Reply by yani66** [reply]
Thank you for your help and sorry for the inconvenience.
